client/src/BidList.js

```diff
@@ -39,90 +40,110 @@ const BuyerTenderList = () => {
 	}, [currentPage, tenderId]);
 
 	const handleBidStatusChange = async (bidId, status) => {
-		const response = await post(`/api/bid/${bidId}/status`, { status });
-		if (response.code === "SUCCESS") {
-			setUpdatedStatus({ id: bidId, status: status });
-		} else {
-			setErrorMsg("Error fetching bids!");
+		try {
+			const response = await post(`/api/bid/${bidId}/status`, { status });
+			if (response.code === "SUCCESS") {
+				setUpdatedStatus(
+					`Updated the status for bid id ${bidId} to ${status} !`
+				);
+				fetchBids(tenderId, currentPage);
+				setIsVisible(true);
+				if (status === "Awarded") {
+					setIsAnyBidAccepted(true);
+				}
+			} else {
+				setStatusError("Error updating bid status!");
+			}
```

Comment:
Please, test your code. The else here will never be reached. If there is an error, an exception is thrown and the catch clause is executed.

```suggestion
				setUpdatedStatus(
					`Updated the status for bid id ${bidId} to ${status}!`
				);
				fetchBids(tenderId, currentPage);
```